completion.go

```diff
@@ -161,7 +161,17 @@ func checkEndpointSupportsModel(endpoint, model string) bool {
 func checkPromptType(prompt any) bool {
 	_, isString := prompt.(string)
 	_, isStringSlice := prompt.([]string)
-	return isString || isStringSlice
+	isInterfaceSlice := false
+	if slice, isSlice := prompt.([]interface{}); isSlice {
+		isInterfaceSlice = true
+		for _, item := range slice {
+			if _, isStringItem := item.(string); !isStringItem {
+				isInterfaceSlice = false
+				break
+			}
+		}
+	}
+	return isString || isStringSlice || isInterfaceSlice
```

Comment:
@AyushSawant18588 thank you for the PR! Could we please use a less nested version of this with early returns? Something like this https://go.dev/play/p/cS9jrYnWrvN

Reply:
Yeah sure I will check and update the PR

Reply:
I have updated the PR @sashabaranov, can you please take a look?